subprojects/plugins/src/main/java/org/gradle/api/internal/java/JavaLibrary.java

```diff
@@ -175,4 +182,30 @@ public String getName() {
             return runtimeDependencies.withType(ModuleDependency.class);
         }
     }
+
+    /**
+     * This is only here to provide backward compatibility for the Shadow plugin. Remove in 5.0.
+     */
+    private static class BackwardsCompatibilityIsolatableFactory implements IsolatableFactory {
+        @Override
+        public <T> Isolatable<T> isolate(final T value) {
+            return new Isolatable<T>() {
+                @Override
+                public T isolate() {
+                    return value;
+                }
+
+                @Nullable
+                @Override
+                public <S> Isolatable<S> coerce(Class<S> type) {
+                    return null;
+                }
+
+                @Override
+                public void appendToHasher(BuildCacheHasher hasher) {
+                    throw new UnsupportedOperationException();
```

Comment:
I guess the plugin could potentially hit this (?) But it does not in the test scenario.... lets go with this.